perllib/FixMyStreet/Cobrand/Peterborough.pm

```diff
@@ -476,26 +511,157 @@ sub image_for_unit {
     return $images->{$service_id};
 }
 
+# XXX
+# Error handling
+# Holidays, bank holidays?
+# Monday limit, Tuesday limit etc.?
+# Check which bulky collections are pending, open
+# Check if collection is free or chargeable
+sub find_available_bulky_slots {
+    my ( $self, $property, $last_earlier_date_str ) = @_;
+
+    my $key
+        = 'peterborough:bartec:available_bulky_slots:'
+        . ( $last_earlier_date_str ? 'later' : 'earlier' ) . ':'
+        . $property->{uprn};
+    return $self->{c}->session->{$key} if $self->{c}->session->{$key};
+
+    my $bartec = $self->feature('bartec');
+    $bartec = Integrations::Bartec->new(%$bartec);
+
+    my $window    = _bulky_collection_window($last_earlier_date_str);
+    if ( $window->{error} ) {
+        # XXX Handle error gracefully
+        die $window->{error};
+    }
+    my $workpacks = $bartec->Premises_FutureWorkpacks_Get(
+        date_from => $window->{date_from},
+        date_to   => $window->{date_to},
+        uprn      => $property->{uprn},
+    );
+
+    my @available_slots;
+
+    my $suffix_date_parser
+        = DateTime::Format::Strptime->new( pattern => '%d%m%y' );
+    my $workpack_date_pattern = '%FT%T';
+    my $workpack_date_parser
+        = DateTime::Format::Strptime->new(
+        pattern => $workpack_date_pattern );
+
+    my $last_workpack_date;
+    for my $workpack (@$workpacks) {
+        # Can be an arrayref or a hashref
+        my $action_data = $workpack->{Actions}{Action};
```

Comment:
It looks like `$workpack->{Actions}` can have an arrayref as a value if there is more than one `{Action}` in which case get the error Not a HASH reference (Edit: which I see you've noted - so I guess it should be my `$action_data = $workpack->{Actions};` and the rest of the code deals with popping it in an arrayref - but it does break [t/app/controller/waste_peterborough.t](https://github.com/mysociety/fixmystreet/pull/4047/files#diff-83fd9cc62337a88c4ff8caeda4a669b3a3717c8b3be8c53100edb9004232e71b))